### DISCUSSION TITLE
use BS in PDF layout output

### DISCUSSION THREAD

**Discussion by wnunezc** [twbs/bootstrap#36766] [General]
Hello, is there a way to create an HTML file and have it have the BS styles and then generate a PDF file and have the design be similar or identical to what is seen on the screen, I can't find a way to do it, if someone Has he made or has a tutorial? I would appreciate it if you pass it on to me, thank you.

I'm looking to do this in two ways:
1. From javascript with the data that is on the screen.
2. from the backend with php.

**Comment by TravelTimN** [comment]
@arcanisgk 
You would need to use something like [jsPDF](https://github.com/MrRio/jsPDF) to convert the page into a PDF.

However, please be aware that just by itself, it cannot retain *[Bootstrap]* styles and formatting. You would need to include a plugin like [html2canvas](https://github.com/niklasvh/html2canvas).

If you don't need to dynamically generate the PDF files, another simple option I've used before is the Chrome Developer Tools.

- PC: Press **Ctrl + Shift + I**, then **Ctrl + Shift P**.
- Mac: **Press Command + Option + I**, then **Command + Shift P**.
- Then type **"screenshot"** to see the four screenshot options.

From there, you can simply save the screenshot locally as a PDF file on your system.

Hope this is helpful in some way for your task!

**Reply by wnunezc** [reply]
thank you for response, any idea to work it in backend i have used mPDF with HTML and in-line style but any test case to. implement BS?

**Reply by TravelTimN** [reply]
Sadly I'm not familiar with PHP, I use more Python backend with Django/Flask/etc.

Hopefully something like [this](https://stackoverflow.com/a/733326) would be helpful though?
It's a bit dated, but seems to still be quite resourceful!

**Reply by wnunezc** [reply]
for the backend I already have something that generates the PDFs... not exactly as described in the example you have given me, I usually use mPDF which is a bit more user-friendly with developments. That answer is just suggestions of what to use... I already have a broad idea about it, what I need is to be able to implement it, I'll post it here because it's related to implementing BS in the PDF output.

**Reply by TravelTimN** [reply]
I see. I'm afraid I wouldn't be able to help much further in that case, since I'm not as familiar with PHP or mPDF.
Hopefully someone else comes along and is able to help.

In the meantime, I found [this](https://stackoverflow.com/questions/49102418/how-to-use-bootstrap-in-mpdf) that might shed some light for your situation. Best of luck! 🍀

**Reply by wnunezc** [reply]
@TravelTimN  Thank you very much, anything you find on the subject update this post.

**Reply by gerardnico** [reply]
You can use also use a web driver that generate a pdf from a page.

**Reply by wnunezc** [reply]
@gerardnico Although what you say solves the problem, you have not given an explanation or an indication of what it means or how to do what you suggest. what you just contributed is similar to saying: **_maybe a developer can solve it._**

**Reply by gerardnico** [reply]
Sorry to take some of my time trying to resolve your problem.

The curve is deep and therefore if you are not willing to take it, just ignore it.

Otherwise, you can find an introduction:
https://datacadamia.com/web/browser/webdriver

Or just search Google, there is plenty of examples.
